sample_metadata/parser/generic_parser.py

```diff
@@ -321,14 +321,19 @@ def parse_manifest(
             )
             added_ext_sample_to_internal_id[new_sample.external_id] = sample_id
 
```

Comment:
I think a simpler solution might be to include the added sample id in the list, eg:

```suggestion
existing_external_id_to_cpgid[new_sample.external_id] = sample_id
```

Otherwise we'd probably want `allow_missing=False` in the sapi call.

Reply:
great point, thanks!